scarb/src/core/manifest/mod.rs

```diff
@@ -161,6 +163,25 @@ pub struct ManifestCompilerConfig {
     pub sierra_replace_ids: bool,
 }
 
+impl ManifestCompilerConfig {
+    pub fn default_dev() -> Self {
+        Self {
+            sierra_replace_ids: true,
+        }
+    }
+    pub fn default_release() -> Self {
+        Self {
+            sierra_replace_ids: false,
+        }
+    }
+}
+
+impl Default for ManifestCompilerConfig {
+    fn default() -> Self {
+        Self::default_dev()
+    }
+}
+
```

Comment:
Wdyt about:
1. Introducing `DefaultForProfile` trait
2. Skipping `impl Default` for all profile-sensitive types